store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/ProxyStore.java

```diff
@@ -143,8 +142,13 @@ public boolean isSupported(final Class<? extends Operation> operationClass) {
         return getSupportedOperations().contains(operationClass);
     }
 
-    protected Set<StoreTrait> fetchTraits() throws OperationException {
-        Set<StoreTrait> newTraits = executeOpChainViaUrl(new OperationChain<>(new GetTraits.Builder().currentTraits(false).build()), new Context());
+    protected Set<StoreTrait> fetchTraits(final Operation operation) throws OperationException {
+        Set<StoreTrait> newTraits;
+        try {
+            newTraits = executeOpChainViaUrl(new OperationChain<>(operation), new Context());
+        } catch (final OperationException e) {
```

Comment:
can the catch go at the end. catch everything and anything please.

Reply:
Done